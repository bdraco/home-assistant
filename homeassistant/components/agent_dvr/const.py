"""Constants for agent_dvr component."""
DOMAIN = "agent_dvr"
SERVERS = "servers"
DEVICES = "devices"
ENTITIES = "entities"
BINARY_SENSOR_SCAN_INTERVAL_SECS = 2
CAMERA_SCAN_INTERVAL_SECS = 5
SERVICE_UPDATE = "update"
SIGNAL_UPDATE_AGENT = "agent_update"
ATTRIBUTION = "Data provided by ispyconnect.com"
SERVER_URL = "server_url"
CONNECTION = "connection"
