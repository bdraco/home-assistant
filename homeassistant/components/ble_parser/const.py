"""Constants for the BLE Parser integration."""

DOMAIN = "ble_parser"
