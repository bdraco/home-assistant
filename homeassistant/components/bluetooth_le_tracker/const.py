"""Constants for the Bluetooth LE Tracker integration."""

DOMAIN = "bluetooth_le_tracker"
CONF_TRACK_BATTERY = "track_battery"
CONF_TRACK_BATTERY_INTERVAL = "track_battery_interval"

DEFAULT_TRACK_BATTERY = False
DEFAULT_TRACK_BATTERY_INTERVAL = 86400
SIGNAL_BLE_DEVICE_NEW = "bluetooth_le_tracker_new_device"
SIGNAL_BLE_DEVICE_UNAVAILABLE = "bluetooth_le_tracker_unavailable_device"
SIGNAL_BLE_DEVICE_SEEN = "bluetooth_le_tracker_seen_device"
SIGNAL_BLE_DEVICE_BATTERY_UPDATE = "bluetooth_le_tracker_battery_update"

ATTR_ADDRESS = "address"
ATTR_RSSI = "rssi"

BLE_PREFIX = "BLE_"
