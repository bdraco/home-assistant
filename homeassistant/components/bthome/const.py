"""Constants for the BTHome Bluetooth integration."""
from __future__ import annotations

from typing import Final, TypedDict

DOMAIN = "bthome"


CONF_DEVICE_KEY: Final = "device_key"
CONF_EVENT_PROPERTIES: Final = "event_properties"
EVENT_PROPERTIES: Final = "event_properties"
EVENT_TYPE: Final = "event_type"
BTHOME_BLE_EVENT: Final = "bthome_ble_event"


class BTHomeBleEvent(TypedDict):
    """BTHome BLE event data."""

    device_id: str
    address: str
    event_type: str
    event_properties: dict[str, str | int | float | None] | None
