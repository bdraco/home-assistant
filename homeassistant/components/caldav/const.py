"""Constants for caldav integration."""
from caldav.lib.error import DAVError
from requests.exceptions import RequestException

CONF_ADD_CUSTOM_CALENDAR = "add_new_custom_calendars"
CONF_CALENDAR = "calendar"
CONF_CALENDARS = "calendars"
CONF_CUSTOM_CALENDARS = "custom_calendars"
CONF_DAYS = "days"
CONF_SEARCH = "search"
DEFAULT_VERIFY_SSL = True
DOMAIN = "caldav"
OFFSET = "!!"
CALDAV_EXCEPTIONS = (RequestException, DAVError)
