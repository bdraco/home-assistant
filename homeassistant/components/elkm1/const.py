"""Support the ElkM1 Gold and ElkM1 EZ8 alarm/integration panels."""

DOMAIN = "elkm1"
