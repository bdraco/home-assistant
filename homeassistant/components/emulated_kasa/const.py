"""Constants for emulated_kasa."""

CONF_POWER = "power"
DOMAIN = "emulated_kasa"
