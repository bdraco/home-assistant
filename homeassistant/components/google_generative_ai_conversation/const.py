"""Constants for the Google Generative AI Conversation integration."""

import logging

DOMAIN = "google_generative_ai_conversation"
LOGGER = logging.getLogger(__package__)
CONF_PROMPT = "prompt"
CONF_TONE_PROMPT = "tone_prompt"
DEFAULT_PROMPT = """This smart home is controlled by Home Assistant.

An overview of the areas and the devices in this smart home:
{%- for area in areas() %}
  {%- set area_info = namespace(printed=false) %}
  {%- for device in area_devices(area) -%}
    {%- if not device_attr(device, "disabled_by") and not device_attr(device, "entry_type") and device_attr(device, "name") %}
      {%- if not area_info.printed %}

{{ area_name(area) }}:
        {%- set area_info.printed = true %}
      {%- endif %}
- {{ device_attr(device, "name") }}{% if device_attr(device, "model") and (device_attr(device, "model") | string) not in (device_attr(device, "name") | string) %} ({{ device_attr(device, "model") }}){% endif %}
    {%- endif %}
  {%- endfor %}
{%- endfor %}
"""

CONF_RECOMMENDED = "recommended"
CONF_CHAT_MODEL = "chat_model"
RECOMMENDED_CHAT_MODEL = "models/gemini-1.5-flash-latest"
CONF_TEMPERATURE = "temperature"
RECOMMENDED_TEMPERATURE = 1.0
CONF_TOP_P = "top_p"
RECOMMENDED_TOP_P = 0.95
CONF_TOP_K = "top_k"
RECOMMENDED_TOP_K = 64
CONF_MAX_TOKENS = "max_tokens"
RECOMMENDED_MAX_TOKENS = 150
