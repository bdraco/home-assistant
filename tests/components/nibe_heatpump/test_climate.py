"""Test the Nibe Heat Pump config flow."""
from typing import Any
from unittest.mock import patch

from nibe.coil_groups import CLIMATE_COILGROUPS, UNIT_COILGROUPS
from nibe.heatpump import Model
import pytest
from syrupy import SnapshotAssertion

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant

from . import MockConnection, async_add_model


@pytest.fixture(autouse=True)
async def fixture_single_platform():
    """Only allow this platform to load."""
    with patch("homeassistant.components.nibe_heatpump.PLATFORMS", [Platform.CLIMATE]):
        yield


@pytest.mark.parametrize(
    ("model", "climate_id", "entity_id"),
    [
        (Model.S320, "s1", "climate.climate_system_s1"),
    ],
)
async def test_basic(
    hass: HomeAssistant,
    mock_connection: MockConnection,
    model: Model,
    climate_id: str,
    entity_id: str,
    coils: dict[int, Any],
    entity_registry_enabled_by_default: None,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setting of value."""
    climate = CLIMATE_COILGROUPS[model.series][climate_id]
    unit = UNIT_COILGROUPS[model.series]["main"]
    if climate.active_accessory is not None:
        coils[climate.active_accessory] = "ON"
    coils[climate.current] = 20.5
    coils[climate.setpoint_heat] = 21.0
    coils[climate.setpoint_cool] = 30.0
    coils[climate.mixing_valve_state] = "ON"
    coils[climate.use_room_sensor] = "ON"
    coils[unit.prio] = "HEAT"
    coils[unit.cooling_with_room_sensor] = "ON"

    await async_add_model(hass, model)

    assert hass.states.get(entity_id) == snapshot(name="1. initial")

    mock_connection.mock_coil_update(unit.prio, "OFF")

    assert hass.states.get(entity_id) == snapshot(name="2. idle")
