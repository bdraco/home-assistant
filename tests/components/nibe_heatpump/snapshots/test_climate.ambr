# serializer version: 1
# name: test_basic[Model.S320-s1-climate.climate_system_s1][1. initial]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'current_temperature': 20.5,
      'friendly_name': 'Climate System S1',
      'hvac_action': <HVACAction.HEATING: 'heating'>,
      'hvac_modes': list([
        <HVACMode.AUTO: 'auto'>,
        <HVACMode.HEAT: 'heat'>,
        <HVACMode.HEAT_COOL: 'heat_cool'>,
      ]),
      'max_temp': 35.0,
      'min_temp': 5.0,
      'supported_features': <ClimateEntityFeature: 3>,
      'target_temp_high': 30.0,
      'target_temp_low': 21.0,
      'target_temp_step': 0.5,
      'temperature': None,
    }),
    'context': <ANY>,
    'entity_id': 'climate.climate_system_s1',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'heat_cool',
  })
# ---
# name: test_basic[Model.S320-s1-climate.climate_system_s1][2. idle]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'current_temperature': 20.5,
      'friendly_name': 'Climate System S1',
      'hvac_action': <HVACAction.IDLE: 'idle'>,
      'hvac_modes': list([
        <HVACMode.AUTO: 'auto'>,
        <HVACMode.HEAT: 'heat'>,
        <HVACMode.HEAT_COOL: 'heat_cool'>,
      ]),
      'max_temp': 35.0,
      'min_temp': 5.0,
      'supported_features': <ClimateEntityFeature: 3>,
      'target_temp_high': 30.0,
      'target_temp_low': 21.0,
      'target_temp_step': 0.5,
      'temperature': None,
    }),
    'context': <ANY>,
    'entity_id': 'climate.climate_system_s1',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'heat_cool',
  })
# ---
